sunpy/map/map_factory.py

```diff
@@ -186,55 +187,58 @@ def _parse_args(self, *args, **kwargs):
         # For each of the arguments, handle each of the cases
         i = 0
         while i < len(args):
-
-            arg = args[i]
-
-            # Data-header or data-WCS pair
-            if isinstance(arg, SUPPORTED_ARRAY_TYPES):
-                arg_header = args[i+1]
-                if isinstance(arg_header, WCS):
-                    arg_header = args[i+1].to_header()
-
-                if self._validate_meta(arg_header):
-                    pair = (args[i], OrderedDict(arg_header))
-                    data_header_pairs.append(pair)
-                    i += 1    # an extra increment to account for the data-header pairing
-
-            # A database Entry
-            elif isinstance(arg, DatabaseEntryType):
-                data_header_pairs += self._read_file(arg.path, **kwargs)
-
-            # Already a Map
-            elif isinstance(arg, GenericMap):
-                already_maps.append(arg)
-
-            # URL
-            elif isinstance(arg, str) and _is_url(arg):
-                url = arg
-                path = str(cache.download(url).absolute())
-                pairs = self._read_file(path, **kwargs)
-                data_header_pairs += pairs
-
-            # File system path (file or directory or glob)
-            elif _possibly_a_path(arg):
-                path = pathlib.Path(arg).expanduser()
-                if _is_file(path):
+            try:
```

Comment:
What was the need to put the entire block of code in the try? Could this not be focused on the part that does the reading of a file?